### PR TITLE
Fix retire script

### PR DESCRIPTION
**Pull request by ahelal**
* Change the default behaviour of init.d script to try retire the worker. if it fails it kills the worker process.
* Make the retire script a helper script
* Change the retire behaviour checks from `fly` to `concourse retire-worker`  
* Make the echo more init.d friendly 

Probably in future release we should remove init.d and use systemd
 
@rnurgaliyev @mazubieta Feedback is highly appreciated

### CODE REVIEW

File: templates/concourse-worker-init.sh.j2 [hunk 1]
```diff
@@ -66,39 +81,46 @@ start() {
 }
 
 stop() {
-    PID=$(PROG_PID)
-    if [ -n "$PID" ]; then
-        echo -n  "Stoping $NAME "
-        kill -9 $PID
-        [ "$?" -ne 0 ] && echo "[ FAILED ]" && exit 1
+    if [ -n "$(PROG_PID)" ]; then
+        echo -n  "Retirirng/Stoppping ${NAME} "
```
Comment (mazubieta) [inline templates/concourse-worker-init.sh.j2]: two small typos here

File: templates/concourse-worker-init.sh.j2 [hunk 2]
```diff
@@ -66,39 +81,46 @@ start() {
...
         fi
     else
-        echo "$NAME not running."
+        echo "${NAME} not running."
     fi
 }
 
+retire() {
+    # loop infinitely till wil retire
+    echo -n  "Retirirng ${NAME} "
```
Comment (mazubieta) [inline templates/concourse-worker-init.sh.j2]: typo

**Comment by ahelal** [comment]
@mazubieta thanks for your input. Historically [start-stop-daemon](https://manpages.debian.org/jessie/dpkg/start-stop-daemon.8.en.html) has a limit on name field 15 chars, but I guess I can change it to `ConcourseWorker`

**Comment by mazubieta** [comment]
Ah ... didn't know that!  Really isn't a big deal.  But that works, if you
care to make that change.

Thanks for your help getting these changes/fixes in!!

**Pull request merged**